tests/fail/stacked_borrows/disable_mut_does_not_merge_srw.rs

```diff
@@ -0,0 +1,17 @@
+// This tests demonstrates the effect of 'Disabling' mutable references on reads, rather than
+// removing them from the stack -- the latter would 'merge' neighboring SRW groups which we would
+// like to avoid.
+fn main() {
+    unsafe {
+        let mut mem = 0;
+        let base = &mut mem as *mut i32; // the base pointer we build the rest of the stack on
+        let mutref = &mut *base;
+        let raw = mutref as *mut i32;
+        // in the stack, we now have [base, mutref, raw]
+        let _val = *base;
+        // now mutref is disabled
```

Comment:
I think this is a pretty good test, but it doesn't cover the exact situation that would help me with building a tag GC. What I'm looking for is a case where the disabled pointer isn't accessible anymore, but must be left in the stack.

This achieves the desired effect; if I configure the GC to run on every machine step in its current (broken) state, it deletes the tag for `mutref` and the test passes:
```diff
diff --git a/tests/fail/stacked_borrows/disable_mut_does_not_merge_srw.rs b/tests/fail/stacked_borrows/disable_mut_does_not_merge_srw.rs
index 1dcd7621..2c3bf808 100644
--- a/tests/fail/stacked_borrows/disable_mut_does_not_merge_srw.rs
+++ b/tests/fail/stacked_borrows/disable_mut_does_not_merge_srw.rs
@@ -5,8 +5,10 @@ fn main() {
     unsafe {
         let mut mem = 0;
         let base = &mut mem as *mut i32; // the base pointer we build the rest of the stack on
-        let mutref = &mut *base;
-        let raw = mutref as *mut i32;
+        let raw = {
+            let mutref = &mut *base;
+            mutref as *mut i32
+        };
         // in the stack, we now have [base, mutref, raw]
         let _val = *base;
         // now mutref is disabled
```
So this diff would be nice, or perhaps one where the pointer only goes out of scope once it is disabled.

Reply:
Whether or not `mutref` is still in scope really shouldn't make a difference... so that test really only makes sense with your GC in mind. But sure I guess we can do that.